File: rust/src/models/matrix_pricing.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::matrix_row::MatrixRow;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MatrixPricing {
    pub rates: Vec<MatrixRow>,
}

impl MatrixPricing {
    pub fn new(rates: Vec<MatrixRow>) -> Self {
        Self { rates }
    }
}

File: typescript/src/models/matrixPricing.ts
// this file is @generated
import { type MatrixRow, MatrixRowSerializer } from "./matrixRow";

export interface MatrixPricing {
  rates: MatrixRow[];
}

export const MatrixPricingSerializer = {
  _fromJsonObject(object: any): MatrixPricing {
    return {
      rates: object["rates"].map((item: any) =>
        MatrixRowSerializer._fromJsonObject(item)
      ),
    };
  },

  _toJsonObject(self: MatrixPricing): any {
    return {
      rates: self.rates.map((item: any) => MatrixRowSerializer._toJsonObject(item)),
    };
  },
};
